File: backend/dotnet/clean-architecture/full/src/Core/App.Template.Application/Features/FileManagement/Commands/UploadFile/UploadFileCommand.cs
using AppTemplate.Application.DTOs;
using MediatR;

namespace AppTemplate.Application.Features.FileManagement.Commands.UploadFile;

public record UploadFileCommand : IRequest<UploadedFileDto>
{
    public required Stream FileStream { get; init; }
    public required string FileName { get; init; }
    public required string ContentType { get; init; }
    public required long FileSize { get; init; }
    public string? Description { get; init; }
    public string? Category { get; init; }
    public bool IsPublic { get; init; }
}

File: backend/nestjs/clean-architecture/minimal/src/modules/file-management/application/commands/upload-file.command.ts
export class UploadFileCommand {
  constructor(
    public readonly fileBuffer: Buffer,
    public readonly originalFileName: string,
    public readonly contentType: string,
    public readonly fileSize: number,
    public readonly description?: string,
    public readonly category?: string,
    public readonly isPublic?: boolean,
  ) {}
}
